ballista/rust/scheduler/src/flight_sql.rs

```diff
@@ -0,0 +1,460 @@
+            let status = self
+                .server
+                .state
+                .task_manager
+                .get_job_status(&job_id)
+                .await
+                .map_err(|e| {
+                    let msg = format!("Error getting status for job {}: {:?}", job_id, e);
+                    error!("{}", msg);
+                    Status::internal(msg)
+                })?;
+            let status: JobStatus = match status {
+                Some(status) => status,
+                None => {
+                    let msg = format!("Error getting status for job {}!", job_id);
+                    error!("{}", msg);
+                    Err(Status::internal(msg))?
+                }
+            };
+            let status: job_status::Status = match status.status {
+                Some(status) => status,
+                None => {
+                    let msg = format!("Error getting status for job {}!", job_id);
+                    error!("{}", msg);
+                    Err(Status::internal(msg))?
+                }
+            };
+            let completed = match status {
+                job_status::Status::Queued(_) => continue,
+                job_status::Status::Running(_) => continue,
+                job_status::Status::Failed(e) => Err(Status::internal(format!(
+                    "Error building plan: {}",
+                    e.error
+                )))?,
+                job_status::Status::Completed(comp) => comp,
+            };
+            let mut fieps: Vec<_> = vec![];
+            for loc in completed.partition_location.iter() {
+                let fetch = if let Some(ref id) = loc.partition_id {
+                    let fetch = protobuf::FetchPartition {
+                        job_id: id.job_id.clone(),
+                        stage_id: id.stage_id,
+                        partition_id: id.partition_id,
+                        path: loc.path.clone(),
+                    };
+                    protobuf::Action {
+                        action_type: Some(protobuf::action::ActionType::FetchPartition(
+                            fetch,
+                        )),
+                        settings: vec![],
+                    }
+                } else {
+                    Err(Status::internal("Error getting partition ID".to_string()))?
+                };
+                let authority = if let Some(ref md) = loc.executor_meta {
+                    format!("{}:{}", md.host, md.port)
+                } else {
+                    Err(Status::internal(
+                        "Invalid partition location, missing executor metadata"
+                            .to_string(),
+                    ))?
+                };
+                if let Some(ref stats) = loc.partition_stats {
+                    num_rows += stats.num_rows;
+                    num_bytes += stats.num_bytes;
+                } else {
+                    Err(Status::internal("Error getting stats".to_string()))?
```

Comment:
There's a lot of untested branches here. I'm still trying to figure out how to test properly. If the Rust implementation of the Flight SQL client is far enough along, integration tests would be good. Otherwise (and perhaps in addition to), unit test would be helpful. But so much of this is interacting with other parts of the system, I'm not quite sure how to do it. I'll look over `test_poll_work()` as a reference.